cost-analyzer/templates/awsstore-deployment-template.yaml

```diff
@@ -25,7 +25,7 @@ spec:
             - image: {{ .Values.awsstore.imageNameAndVersion }}
               name: awsstore
               # Just sleep forever
-              command: [ "sleep" ]
-              args: [ "infinity" ]
+              command: [ "/bin/sh" ]
+              args: [ "-c", "while true; do go/bin/app; sleep infinity; done" ]
```

Comment:
I wonder if we need to specify command+args at all here; I feel like this should be the container's responsibility unless we're doing something non-standard

Reply:
I thought the same thing the only input I can give is that without specifying an arg here I noticed nothing in the pod log. Specifying this arg at least showed:
`I0228 18:05:56.885938      11 main.go:28] Read region from metadata us-east-2`
`I0228 18:05:56.886008      11 main.go:38] Using productcode 8c0{redacted}9c`

That could just be on my side though.

Reply:
That being said, it just hangs at those two lines, and the dashboard doesnt update to show that I am on the enterprise license. 

I know the pod has the proper IRSA permissions cause I can do the call by installing the aws-cli on the container and get back the JWT using the product code given to be from the go app's logs.

Reply:
Hi @wesleyorama-grindr, sorry for the late response! Which dashboard doesn't update to show the enterprise license? Nothing in kubecost automatically checks to assert that we've connected to the aws marketplace. We should just be using the marketplace for billing and redemption. You would want to just pop in the enterprise key that your account executive contact provided on setup here: https://github.com/kubecost/cost-analyzer-helm-chart/blob/develop/cost-analyzer/values.yaml#L769

Reply:
That is to say-- if you see those logs being produced and the awsstore pod being run forever, this _should_ be working.

Reply:
hey @AjayTripathy, the settings page of Kubecost still shows the non-enterprise version of the software running. 

I got here by following the instructions from the AWS Marketplace. I did end up using the ent key, but since there is documentation showing people to do this... Maybe removing that from the marketplace templates and instead just pointing to the docs on using the ent key provided in the onboarding email would be better. Then we could clean this code up. 

Going back to the logs, I would only get those two logs and no more. The pod would run forever, but it would run forever without the sleep. Again without being able to look at the actual files that make up the image I can only assume its hanging trying to work with the marketplace. Like I said I verified IRSA permissions from that container so I know that shouldn't be a problem. 

Ultimately I have resolved my issue of not being able to use my license by using the ent key provided to me. However this process is still broken. If y'all decide to keep this process in place, I am happy to help troubleshooting and continuing to work on it.

Let me know what y'all think.

Reply:
This is the the instructions provided on the marketplace entitlement:
![image](https://user-images.githubusercontent.com/83981785/156233278-1e0f6879-170d-4e06-9c86-1b7b60df201f.png)

Here is that link for convenience:
https://github.com/kubecost/docs/blob/master/aws-marketplace-install.md

Reply:
Hey Wesley, I'm so sorry that the docs were out of date.  Your point about removing the marketplace template until the process is updated is noted, and thank you for the feedback. We're currently using the AWS marketplace just for payment processing-- fulfillment is handled by us and the enterprise key.  

I've reached out to the team to re-visit those docs and help you all get a more detailed onboarding meeting with our current process for enterprise onboarding to hopefully smooth this out for the next users.